### PR TITLE
Working with coroutine in python

### LINKED ISSUE #4759: Working with coroutine in python
#### Proposed title of the article
[Languages] Working with coroutine in python

#### Proposed article introduction
Coroutines are subroutines that have been expanded. They are used for cooperative multitasking, in which a process voluntarily yields (gives up) control on a periodic or idle basis to allow other programs to execute at the same time.
Coroutines are similar to generators in Python, but with a few extra methods and subtle differences in how yield statements are used. Coroutines can consume data as generators produce data for iteration.
The reader should have prior knowledge of Generators in python.

#### Key takeaways 
By the end of this tutorial, the reader should be able to understand:
- What are coroutines in python
- The difference between coroutine and subroutine
- How to execute and close coroutines in python
- Chaining coroutines

#### Article quality
This will be an in-depth tutorial in terms of coverage of the concept. I will also cover how coroutine is different from threads and To make it more practicable I will include examples relevant and to be well understood by the reader including explanations to enhance understanding. This article will include an easy but understandable language making it beginner-friendly.

#### References
N/A

### PR DESCRIPTION
**Pull request by Nganga89**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4759

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Please always use the template provided by us.
I edited your initial comment and added the template. Check what matches your article.

**Comment by Nganga89** [comment]
> Please always use the template provided by us. I edited your initial comment and added the template. Check what matches your article.

Noted with thanks. I have made the correction on that.

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/articles/working-with-coroutine-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+### Introduction
+Expanded subroutines are referred to as coroutines. A process willingly relinquishes control on a periodic or idle basis to enable other programs to run at the same time, which is known as cooperative multitasking.
+
+With a few more methods and slight variations in yield statements, coroutines vary from Python's generators. Data may be consumed by coroutines in the same way that data is generated for iteration.
+
```
Comment (geoffrey45) [inline content/articles/working-with-coroutine-in-python/index.md:5]: Explain what Python generators are, before contrasting them with coroutines.

File: content/articles/working-with-coroutine-in-python/index.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+### How different are coroutines from threads?
+- With threads, the operating system changes running threads proactively as per its scheduler, which is an algorithm in the operating system kernel.
+- Activities are collaboratively multi-tasked by stopping and restarting functions at predetermined times, it can sometimes be within a single thread. This is accomplished through the use of coroutines.
+
+### Coroutines in a Python program
+Coroutines will use the `send()` method to transmit names, and only those with the prefix "Student" will be displayed at the end.
+
+```python
+def print_cognomen (prfx):
```
Comment (geoffrey45) [inline content/articles/working-with-coroutine-in-python/index.md:38]: Please indent your code blocks using the proper Python standards.

**Review by geoffrey45 — COMMENTED** [review]
@Nganga89 Please check the comments below.

**Comment by Nganga89** [comment]
Hello @geoffrey45, I have done the changes you requested. Thank you.

File: content/articles/working-with-coroutine-in-python/index.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+### Coroutines execution
+Python coroutines and Python generators have a lot in common, as do many other types of Python code. Unless and until the coroutine is called in a program, nothing else happens.
+
+Execution begins when `__next__()` is invoked. The initial expression is performed when invoked.
+
+The coroutine's execution comes to a halt as it awaits the value to be sent to the coroutine object. Each time a new coroutine object receives an initial value, it first checks to see whether the specified prefix is present before printing the name with that prefix.
+
+For as long as you're using the this phrase, you'll keep seeing it until you run across the `name = (yield)` expression again.
+
```
Comment (geoffrey45) [inline content/articles/working-with-coroutine-in-python/index.md:98]: @Nganga89 Please clearly outline what you are referring to by "this" in `this phrase` and "it" in `seeing it`.

**Comment by Nganga89** [comment]
Hello, @geoffrey45 I have done the changes you requested. Thank you.

**Comment by WanjaMIKE** [comment]
@Nganga89 You need to work more on grammar and sentence structure to improve in your future article.

**Comment by WanjaMIKE** [comment]
Merged #4902 into master

**Pull request merged**